### PR TITLE
adding TencentOS Server 4.2

### PR DESCRIPTION
**Pull request by yoav-orca**

### CODE REVIEW

File: ts/server/4.2
Comment (SteveLauC) [inline ts/server/4.2:1]: The directory should be named "tencentos" and this file should be put under it.

Please refer to: https://github.com/which-distro/os-release?tab=readme-ov-file#directory-and-file-naming-conventions

**Comment by SteveLauC** [comment]
Thanks for the PR! Just one naming issue:)

**Comment by yoav-orca** [comment]
@SteveLauC fixed

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**